### DISCUSSION TITLE
Bug when saving to stream

### DISCUSSION THREAD

**Discussion by StepanHusa** [ClosedXML/ClosedXML#2135] [General]
Hi,

I think I found a bug.
When saving a newly created workbook to a stream, there is no option to choose the SpreadsheetDocumentType. This results in a file that has to be repaired by Excel application before opening. 

When saving to file it is deduced from the extension and works ok.

Thanks for making this library the best.

**Comment by StepanHusa** [comment]
It seems like the _spreadsheetDocumentType is not initialized CreatePackage(stream, false, _spreadsheetDocumentType, options) gets unknown type.

I guess the best way to repair it would be to add a SaveAs method to a XLWorkbook class with a SpreadsheetDocumentType parameter:

        public void SaveAs(Stream stream, SpreadsheetDocumentType spreadsheetType, SaveOptions options)
        {
            _spreadsheetDocumentType = spreadsheetType;
            SaveAs(stream, options);
        }

(or something in this direction)

**Reply by joshhughess** [reply]
I found this issue as well and managed to fix it by ensuring that the MemoryStream position was being set to 0 when returning the file for download, not sure if this will help in your situation, but thought I would add a comment to try and provide other context
